tests/chain_tests/atomicswap/atomicswap_operations_tests.cpp

```diff
@@ -322,3 +322,120 @@ SCORUM_TEST_CASE(redeem_by_participant_check)
 }
 
 BOOST_AUTO_TEST_SUITE_END()
+
+struct expired_contract_refund_visitor
+{
+    typedef void result_type;
+
+    database& _db;
+
+    std::map<account_name_type, asset> refund_map;
+
+    expired_contract_refund_visitor(database& db)
+        : _db(db)
+    {
+    }
+
+    void operator()(const expired_contract_refund_operation& op)
+    {
+        refund_map.insert(std::make_pair(op.owner, op.refund));
+    }
+
+    template <typename Op> void operator()(Op&&) const
+    {
+    } /// ignore all other ops
+};
+
+struct atomicswap_expired_contract_fixture : public atomicswap_operation_create_contract_fixture
+{
+};
+
+BOOST_FIXTURE_TEST_SUITE(atomicswap_expired_contract_check, atomicswap_expired_contract_fixture)
+
+SCORUM_TEST_CASE(check_expired_contracts_refund)
+{
+    const account_object& alice = account_service.get_account("alice");
+    const account_object& bob = account_service.get_account("bob");
+
+    BOOST_REQUIRE_THROW(atomicswap_service.get_contract(alice, bob, alice_secret_hash), fc::exception);
+    BOOST_REQUIRE_THROW(atomicswap_service.get_contract(bob, alice, alice_secret_hash), fc::exception);
+
+    signed_transaction tx;
+    tx.set_expiration(db.head_block_time() + SCORUM_MAX_TIME_UNTIL_EXPIRATION);
+    tx.operations.push_back(initiate_op);
+    tx.operations.push_back(participate_op);
+
+    BOOST_REQUIRE_NO_THROW(tx.sign(m_alice_private_key, db.get_chain_id()));
+    BOOST_REQUIRE_NO_THROW(tx.validate());
+
+    BOOST_REQUIRE_NO_THROW(db.push_transaction(tx, default_skip));
+
+    BOOST_REQUIRE_NO_THROW(validate_database());
+
+    expired_contract_refund_visitor visitor(db);
+    db.post_apply_operation.connect([&](const operation_notification& note) { note.op.visit(visitor); });
+
+    auto alice_balance_before_refund = db.obtain_service<dbs_account>().get_account("alice").balance;
+    auto bob_balance_before_refund = db.obtain_service<dbs_account>().get_account("bob").balance;
+
+    // after 1000 blocks all contracts will be expired
+    generate_block(0, initdelegate.private_key, 1000);
```

Comment:
Use generate_blocks(fc::time_point_sec timestamp, bool miss_intermediate_blocks = true)
instead

Reply:
generate_blocks(db.head_block_time() + std::max(SCORUM_ATOMICSWAP_INITIATOR_REFUND_LOCK_SECS, SCORUM_ATOMICSWAP_PARTICIPANT_REFUND_LOCK_SECS)) due to the your test logic